Sources/ConcordiumSwiftSDK/Client.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+import GRPC
+import NIOPosix
+import NIOCore
+
+typealias BlockHash = Data
+
+struct CryptographicParameters {
+    let onChainCommitmentKey: String
+    let bulletproofGenerators: String
+    let genesisString: String
+}
+
+class Client {
+    let grpc: Concordium_V2_QueriesNIOClient
+    
+    init(channel: GRPCChannel) {
+        grpc = Concordium_V2_QueriesNIOClient(channel: channel)
+    }
+    
+    private func getBlockHashInput(blockHash: BlockHash?) -> Concordium_V2_BlockHashInput {
+        if let blockHash {
+            var h = Concordium_V2_BlockHash()
```

Comment:
Since a lot/all? the generated Swift structs have an empty initializer we are prevented from setting the fields in the constructor, so I have added a Kotlin-like 'with' function to allow us to do it in the following way instead. 
@bisgardo Thoughts? While the suggested approach feels less 'old school', I would like your take on whether it adversely affects the readability. 

```suggestion
            if let blockHash {
            return with(Concordium_V2_BlockHashInput()) {
                $0.given = with(Concordium_V2_BlockHash()) {
                    $0.value = blockHash
                }
            }
        } else {
            return with(Concordium_V2_BlockHashInput()) {
                $0.lastFinal = Concordium_V2_Empty()
            }
        }
```

Reply:
Honestly I find that harder to read. Maybe it's just that I'm not used to the pattern.

Agree that setting on a mutable value is ugly though. An alternative would be to add our own wrapper structs with proper initializers. How does that sound to you?

Reply:
It depends :)
If we end up creating wrapper types for most/all of the generated types, then it's probably too much duplication to avoid our sense of aesthetics being violated :D 

However, if we are in fact only exposing a small subset of types, then it might even have a beneficial effect on understandability if we add a "Types" folder with the key types. 

You probably know better than me which of these scenarios will apply :)

Reply:
.. and the 'with' construct has been deleted again